hs_core/serialization.py

```diff
@@ -171,6 +172,8 @@ def create_resource_from_bag(bag_content_path, preserve_uuid=True):
 
     try:
         rm.write_metadata_to_resource(resource)
+        # Force bag files to be re-written
+        create_bag_files(resource)
```

Comment:
@hyi This is problematic because write_metadata_to_resource() is called on derived classes, who call super.write_metadata_to_resource().  The super class's write_metadata_to_resource() calls create_bag_files(), then the bag files will miss the metadata added from the derived class's write_metadata_to_resource().  We could also call create_bag_files() from the end of each derived class's write_metadata_to_resource(), however if in the future an implementer forgets to do this we will have a regression.  Calling create_bag_files() here in create_resource_from_bag() avoids this problem.

For our migration case, I suggest your code follow the same pattern and call create_bag_files() after write_metadata_to_resource().  Does that sound alright to you?